toolchains/solidity/core/crates/slither-server/src/main.rs

```diff
@@ -65,99 +113,179 @@ impl LanguageServer for Backend {
-            Ok(res) => {
-                self.client
-                    .log_message(
-                        MessageType::INFO,
-                        format!(
-                            "File '{}' did generate {} security diagnostics.",
-                            uri.path(),
-                            res.len()
-                        ),
-                    )
-                    .await;
-                self.client.publish_diagnostics(uri, res, None).await;
+    fn new(client: Client) -> Self {
+        Self {
+            client,
+            data: Mutex::new(SlitherData::new()),
+            join_handle: Arc::new(Mutex::new(None)),
+        }
+    }
+
+    async fn analyze_file(&self, file: Url) {
+        if self.is_in_libs(file.path()).await
+        || self.is_in_tests(file.path()).await
+        || !self.is_in_src(file.path()).await
+      {
+          self.client
+              .log_message(
+                  MessageType::INFO,
+                  format!(
+                      "File '{}' is not a source solidity code file, skipping analysis.",
+                      file.path()
+                  ),
+              )
+              .await;
+          return;
+      }
+      self.launch_slither(file).await
+    }
+
+    async fn is_in_libs(&self, path: &str) -> bool {
+        let state = self.data.lock().await;
+        for lib in state.libs_paths.iter() {
+            let fsrc = format!("/{}/", lib.replace("\"", ""));
+            eprintln!("Check path: '{}' contains lib: '{}'", path, fsrc);
+            if path.contains(&fsrc) {
+                return true;
             }
-            Err(SlitherError::ParsingFailed(e)) => {
-                self.client
-                    .log_message(
-                        MessageType::ERROR,
-                        format!(
-                            "File '{}' did generate an error while parsing the output: {:?}",
-                            uri.path(),
-                            e
-                        ),
-                    )
-                    .await;
-                self.client.publish_diagnostics(uri, vec![], None).await;
+        }
+        false
+    }
+
+    async fn is_in_src(&self, path: &str) -> bool {
+        let state = self.data.lock().await;
+        for src in state.src_paths.iter() {
+            let fsrc = format!("/{}/", src.replace("\"", ""));
+            eprintln!("Check path: '{}' contains src: '{}'", path, fsrc);
+            if path.contains(&fsrc) {
+                return true;
             }
-            Err(e) => {
-                self.client
-                    .log_message(
-                        MessageType::ERROR,
-                        format!("File '{}' did generate an error: {:?}", uri.path(), e),
-                    )
-                    .await;
+        }
+        false
+    }
+
+    async fn is_in_tests(&self, path: &str) -> bool {
+        if path.ends_with(".t.sol") {
+            return true;
+        }
```

Comment:
quite useless as you already checks the path

Reply:
Well, and what if some one is a nasty little dev who's tests are in the src folder ?

Reply:
don't care foundry doesn't support it so are we